src/Infrastructure.EntityFramework/Repositories/DatabaseContext.cs

```diff
@@ -177,6 +177,20 @@ protected override void OnModelCreating(ModelBuilder builder)
     // Make sure this is called after configuring all the entities as it iterates through all setup entities.
     private void ConfigureDateTimeUtcQueries(ModelBuilder builder)
     {
+        ValueConverter<DateTime, DateTime> converter;
+        if (Database.IsNpgsql())
+        {
+            converter = new ValueConverter<DateTime, DateTime>(
+                v => v,
+                d => new DateTimeOffset(d).UtcDateTime);
```

Comment:
No it is not, this is just forcing it to actually read the date correct, before it would read a date like such: `2023-08-31T16:29:08.360081+00:00` and with the previous converter would read it as `12:29:08.3600810` (`DateTime.TimeOfDay`) it would still show it as UTC though, it was very weird. This just forces it to actually come back both with it's Kind being UTC and having the same date. 

Regarding you worrying about precision, I think I probably mislead you, the comparer to care less about Millisecond precision isn't actually for Postgres, it's for SQL Server, in all the EF providers we actually have better precision. See data:

```
SQL Server
C#: 16:20:30.2837920
SQL: 16:20:30.2833333

Postgres (Before Change)
C#: 16:29:08.3600810
SQL: 12:29:08.3600810

In DB: 2023-08-31T16:29:08.360081+00:00

Postgres (After Change)
C#: 16:22:17.4456610
SQL: 16:22:17.4456610

MySQL
C#: 16:23:50.4664000
SQL: 16:23:50.4664000

Sqlite
C#: 16:24:22.9868980
SQL: 16:24:22.9868980
```

The C# data point is just a `DateTime.UtcNow` created date and me taking the `TimeOfDay` property from the debug window. The SQL data point is looking at a DateTime that has been read from the database. The misconception is that I changed how postgres handles date and we now have less precision but it's really that I fixed postgres and I can finally actually compare dates but I need to be more lax so that SQL Server tests don't fail. SQL Server does have _some_ ms precision but not as much as C# so I can change this to round the milliseconds a bit and compare those but it seemed like to much math just for test comparisons.

Reply:
Excellent detail, thanks! Deserves to be in the PR description.

I don't love that our tests can't showcase enough precision (trusting that statement) but so be it.

Should this be labelled as needing QA?